engine/src/main/java/org/camunda/bpm/engine/impl/batch/AbstractBatchJobHandler.java

```diff
@@ -169,6 +171,13 @@ protected void createJobEntities(BatchEntity batch, T configuration, String depl
       ByteArrayEntity configurationEntity = saveConfiguration(byteArrayManager, jobConfiguration);
 
       JobEntity job = createBatchJob(batch, configurationEntity);
+
+      if (jobConfiguration.getIds() != null && jobConfiguration.getIds().size() == 1) {
+        job.setProcessInstanceId(jobConfiguration.getIds().get(0));
+        String processDefinitionId = JsonUtil.getString(JsonUtil.asObject(batch.getConfigurationBytes()), PROCESS_DEFINITION_ID, null);
```

Comment:
❌ This implementation will work only for restart batch operations where we store the process definition . We need solution that should work for all batches.

Reply:
[Yana](@yanavasileva),
Thank you for your feedback. Let me check and get back to you

Reply:
[Yana](@yanavasileva ), 
Just out of curiosity, why do we want the property to be set for all the batch types and not just restart? 
For tracking the process instances affected by a batch restart we would only require this property to be set in restart batches right?

Reply:
>why do we want the property to be set for all the batch types and not just restart?

The ticket that this issue refers is subtask of https://github.com/camunda/camunda-bpm-platform/issues/4199.
The end goal is to display batch operation details to affected process instances (and definitions where applicable) for **all batch types**.
To give you more context, there's a separate ticket that you also showed interest in working on that's specific for the restart batch operations: https://github.com/camunda/camunda-bpm-platform/issues/4201